copy-to.js

```diff
@@ -30,50 +30,51 @@ CopyStreamQuery.prototype._detach = function() {
   this.connection.stream.resume()
 }
 
-
 CopyStreamQuery.prototype._transform = function(chunk, enc, cb) {
   var offset = 0
+  var Byte1Len = 1;
+  var Int32Len = 4;
   if(this._remainder && chunk) {
     chunk = Buffer.concat([this._remainder, chunk])
   }
-  if(!this._copyOutResponse) {
-    this._copyOutResponse = true
-    if(chunk[0] == code.ErrorResponse) {
-      this._detach()
-      this.push(null)
-      return cb();
-    }
-    if(chunk[0] != code.CopyOutResponse) {
-      this.emit('error', new Error('Expected CopyOutResponse code (H)'))
-    }
-    var length = chunk.readUInt32BE(1)
-    offset = 1
-    offset += length
-  }
-  while((chunk.length - offset) > 5) {
+
+  var length;
+  var messageCode;
+  var needPush = false;
+
+  while((chunk.length - offset) > (Byte1Len + Int32Len)) {
     var messageCode = chunk[offset]
-    //complete or error
-    if(messageCode == code.CopyDone || messageCode == code.ErrorResponse) {
+    //console.log(c, w, offset, 'PostgreSQL message ' + String.fromCharCode(messageCode))
+    if(!this._gotCopyOutResponse && messageCode === code.CopyOutResponse) {
```

Comment:
instead of `if / else if` statements this might be nice to unroll into a switch statement. AFAIK switches on numbers in v8 are very fast, and particularly on line `52` you could use fall through behavior on the switch instead of doing an `indexOf` on an array every time.